#pragma execution_character_set("UTF-8")//用于qt的编码，如果没有，界面会有中文乱码
#include "qpushbutton.h"
#include <QtWidgets/QMainWindow>
#include "../Class_File/UserClass.h"
#include "../Class_File/SearchClass.h"
#include "../Header_Files/Search.h"
#include "../Header_Files/ApplyFor.h"
#include "qstring.h"

using namespace std;


Search::Search(UserClass nowUser,QWidget* parent)
    : QMainWindow(parent)
{
    ui.setupUi(this);
    this->ButtonApplyFor = ui.ButtonApplyFor;
    this->ButtonClearnText = ui.ButtonClearnText;
    this->ButtonExit = ui.ButtonExit;
    this->ButtonToSearch = ui.ButtonToSearch;
    this->LabelUserName = ui.LabelUserName;
    this->LabelCertState = ui.LabelCertState;
    this->LabelCert = ui.LabelCert;
    this->LabelCertID = ui.LabelCertID;
    this->LabelClientName = ui.LabelClientName;
    this->ComboBoxSearchKind = ui.ComboBoxSearchKind;
    this->TextEditSearchContent = ui.TextEditSearchContent;
    this->TextEditCert = ui.TextEditCert;
    this->LineEditCertID = ui.LineEditCertID;
    this->LineEditClientName = ui.LineEditClientName;
    this->LineEditCertState = ui.LineEditCertState;

    this->LabelCert->hide();
    this->LabelCertID->hide();
    this->LabelClientName->hide();
    this->LabelCertState->hide();
    this->LineEditCertID->hide();
    this->LineEditClientName->hide();
    this->LineEditCertState->hide();
    this->TextEditCert->hide();


    this->lab = ui.label;   //用于测试的label
    this->NowUser = nowUser;

    this->LabelUserName->setText(QString::fromStdString("用户：" + NowUser.UserName));

    QStringList str;
    str << "通过证书号查询" << "通过证书字符串查询" << "";
    ComboBoxSearchKind->insertItems(3, str);//填充下拉菜单

    connect(ui.ButtonToSearch, SIGNAL(clicked()), this, SLOT(ClickSearchButton()));//将按钮和点击事件绑定
    connect(ui.ButtonExit, SIGNAL(clicked()), this, SLOT(ClickEixtButton()));
    connect(ui.ButtonApplyFor, SIGNAL(clicked()), this, SLOT(ClickApplyForButton()));
    connect(ui.ButtonClearnText, SIGNAL(clicked()), this, SLOT(ClickClearnTextButton()));
}


//内容不少。。。。
void Search::ClickSearchButton() {

    string SearchContent = this->TextEditSearchContent->toPlainText().toStdString();
    

    SearchClass Sear1 = SearchClass(SearchContent, SearchKind[this->ComboBoxSearchKind->currentIndex()], 1);
    //SearchClass Sear2 = SearchClass(SearchContent, "CertID", 1);
    Sear1.toSearch();


    this->LabelCert->show();
    this->LabelCertID->show();
    this->LabelClientName->show();
    this->LabelCertState->show();
    this->LineEditCertID->show();
    this->LineEditClientName->show();
    this->LineEditCertState->show();
    this->TextEditCert->show();


    this->LineEditCertID->setText(QString::fromStdString(Sear1.certificateTable[0].CertID));

    this->TextEditCert->setText(QString::fromStdString(Sear1.certificateTable[0].Certificate));

    this->LineEditClientName->setText(QString::fromStdString(Sear1.certificateTable[0].ClientName)); 

    
    //this->LineEditCertState->setText(QString::fromStdString(Sear1.certificateTable[0].CertID));


    //}





    this->lab->setText(this->TextEditSearchContent->toPlainText());

}



void Search::ClickApplyForButton() {

    this->hide();
    //UserClass u=this->NowUser;
    ApplyFor* applyForWin = new ApplyFor(this->NowUser,this);
    connect(applyForWin, SIGNAL(sendsignalApplyFor()), this, SLOT(ReShowWindow()));//当点击子界面EixtButton，调用主界面的ReShowWindow()函数
    applyForWin->show();

}


void Search::ClickClearnTextButton() {
    this->TextEditSearchContent->setText("");
}


void Search::ReShowWindow() {
    this->show();
}

void Search::ClickEixtButton() {

    emit sendsignal();//给父界面传递被关闭信息
    this->close();
}


void Search::closeEvent(QCloseEvent* event) {

    emit sendsignal(); // 给父界面传递被关闭信息

}



